src/metaswitch/clearwater/etcd_shared/common_consul_synchronizer.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python
+
+# Copyright (C) Metaswitch Networks 2017
+# If license terms are provided to you in a COPYING file in the root directory
+# of the source code repository by which you are accessing this code, then
+# the license outlined in that COPYING file applies to your use.
+# Otherwise no rights are granted except for those provided to you by
+# Metaswitch Networks in a separate written agreement.
+
+import consul
+from time import sleep
+import logging
+from metaswitch.common import utils
+from metaswitch.clearwater.etcd_shared.common_synchronizer import CommonSynchronizer
+
+_log = logging.getLogger(__name__)
+
+class CommonConsulSynchronizer(CommonSynchronizer):
```

Comment:
Not clear from comments how `ConsulSynchronizer` and `CommonConsulSynchronizer` differ - and their similar names possibly obfuscate the class hierarchy.  Are you able to add an extra sentence or two??